client/client.go

```diff
@@ -2,21 +2,141 @@ package client
 
 import (
 	"context"
+	"net/http"
 
 	"github.com/coreos/etcd-operator/pkg/spec"
+
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/runtime/schema"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
 )
 
 // Operator operators etcd clusters atop Kubernetes.
 type Operator interface {
-	// Create creates an etcd cluster with the given spec.
+	// Create creates an etcd cluster.
 	Create(ctx context.Context, name string, spec *spec.ClusterSpec) error
 
-	// Delete deletes the etcd cluster with given name
+	// Delete deletes the etcd cluster.
 	Delete(ctx context.Context, name string) error
 
 	// Update updates the etcd cluster with the given spec.
 	Update(ctx context.Context, name string, spec *spec.ClusterSpec) error
 
-	// Status gets the status of the etcd cluster with the given name.
-	Status(ctx context.Context, name string) (error, *spec.ClusterStatus)
+	// Get gets the etcd cluster information.
+	Get(ctx context.Context, name string) (*spec.EtcdCluster, error)
+}
+
+type operator struct {
+	tprClient *rest.RESTClient
+	tprName   string
+	ns        string
+}
+
+func NewOperator(namespace, tpr string) (Operator, error) {
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	configureClient(cfg)
+
+	tprclient, err := rest.RESTClientFor(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &operator{
+		tprClient: tprclient,
+		tprName:   tpr,
+		ns:        namespace,
+	}, nil
+
+}
+
+func (o *operator) Create(ctx context.Context, name string, cspec *spec.ClusterSpec) error {
+	cluster := &spec.EtcdCluster{
+		ObjectMeta: api.ObjectMeta{
```

Comment:
@hongchaodeng If I understand this correctly, /api maps to k8s internal types. v1 will eventually get converted to api types when the requests reach k8s api server. so we should only use v1 externally. 

so i am not sure if it is correct to change ALL api to v1 as you suggested. since the code i mentioned is provided by client-go. i am not sure if i should treat client type registration external or internal. 

i keep what client-go suggests for now.